adobe/react-spectrum
# TagGroup: Nest .spectrum-Tag-action class to prioritize over ClearButton css

No ticket. Noticed it while adding TagGroup into Inventory. There's no priority between `.spectrum-Tag-action` and `.button_spectrum-ClearButton` so in Inventory, `.button_spectrum-ClearButton` styles are applied first and as a result end up higher in hierarchy which we do not want.

I moved `.spectrum-Tag-action` into `.spectrum-Tags-item` so it has a place within the tag class hierarchy. 

## ✅ Pull Request Checklist:


## 📝 Test Instructions:

1. Open any TagGroup story that has isRemovable=true and confirm no visual changes.
2. Open inspector and inspect the `div` that has the `.button_spectrum-ClearButton` class. Note the `width` and `height` css props are crossed out (see screenshot for reference)
<img width="592" alt="image" src="https://user-images.githubusercontent.com/20674023/181386526-ec9418b6-ddf4-4d86-a1ed-9d30a1d28f0c.png">

## 🧢 Your Project:

@quarry/inventory
